external-authority-sources.md

```diff
@@ -0,0 +1,275 @@
+
+Example:
+```json
+{
+  "_embedded": {
+    "externalsources": [
+      {
+        "id": "orcid",
+        "name": "orcid",
+        "hierarchical": false,
+        "type": "externalsource",
+        "_links": {
+          "entries": {
+            "href": "https://dspace7-internal.atmire.com/server/api/integration/externalsources/orcid/entries"
+          },
+          "self": {
+            "href": "https://dspace7-internal.atmire.com/server/api/integration/externalsources/orcid"
+          }
+        }
+      },
+      {
+        "id": "ciencia",
+        "name": "ciencia",
+        "hierarchical": false,
+        "type": "externalsource",
+        "_links": {
+          "entries": {
+            "href": "https://dspace7-internal.atmire.com/server/api/integration/externalsources/ciencia/entries"
+          },
+          "self": {
+            "href": "https://dspace7-internal.atmire.com/server/api/integration/externalsources/ciencia"
+          }
+        }
+      },
+      {
+        "id": "my_staff_db",
+        "name": "my_staff_db",
+        "hierarchical": false,
+        "type": "externalsource",
+        "_links": {
+          "entries": {
+            "href": "https://dspace7-internal.atmire.com/server/api/integration/externalsources/my_staff_db/entries"
+          },
+          "self": {
+            "href": "https://dspace7-internal.atmire.com/server/api/integration/externalsources/my_staff_db"
+          }
+        }
+      }
+    ]
+  },
+  "_links": {
+    "self": {
+      "href": "https://dspace7-internal.atmire.com/server/api/integration/externalsources"
+    }
+  },
+  "page": {
+    "size": 20,
+    "totalElements": 3,
+    "totalPages": 1,
+    "number": 0
+  }
+}
+```
+
+## Single Authority
+**/api/integration/externalsources/<:source-name>**
+
+Provide detailed information about a specific external source. The JSON response document is as follow
+```json
+{
+  "id": "orcid",
+  "name": "orcid",
+  "hierarchical": false,
+  "type": "externalsource"
+}
+```
+
+Exposed links:
+* entries: the list of values managed by the external source
+
+## Linked entities
+### external source entries
+**/api/integration/externalsources/<:source-name>/entries**
+
+It returns the filtered entries managed by the externally, see below 
+
+The supported parameters are:
+* page, size [see pagination](README.md#Pagination) if supported by the external source
+* query: the terms, keywords or prefix to search: mandatory
+* parent: the key of the parent authority when searching in a hierarchical authority 
+
+It returns the entries in the external source matching the query
+
+sample for an external source /server/api/integration/externalsources/orcid/entries?query=Smith&size=2 
+```json
+{
+  "_embedded": {
+    "externalSourceEntries": [
+      {
+        "id": "0000-0002-4271-0436",
+        "display": "Smith, Dean",
+        "value": "Smith, Dean",
+        "metadata": {
+            "dc.identifier.orcid": [
```

Comment:
Something I'm just realizing now.  I suspect these various `metadata` examples will need updating in the future. Depending on the external source you are querying, it may not even return Dublin Core.  Is the plan to somehow pre-map it into Dublin Core before returning it in JSON?  Or will these JSON responses be different per external source?

Reply:
This was clarified in today's Entities Meeting.  We would need to map the responses from the external source into DSpace schemas in order to display results in the Angular UI (and also for later import as well).  So, the idea here is that, in this endpoint, all external sources should look like they are returning DSpace metadata...as the real responses will be mapped to DSpace metadata so they can be returned in our JSON responses..